store/tikv/split_region.go

```diff
@@ -182,12 +183,6 @@ func (s *tikvStore) batchSendSingleRegion(bo *Backoffer, batch batch, scatter bo
 			zap.Stringer("at", kv.Key(batch.keys[i])),
 			zap.Stringer("new region left", logutil.Hex(r)),
 			zap.Error(err))
-		if batchResp.err == nil {
```

Comment:
We cannot ignore this error, otherwise, the caller will think this region is scattering and wait for a response.

I have seen this error in log when I run a big txn test with one-node TiKV.

https://github.com/pingcap/tidb/blob/0df3b74f55f8f8fbde39bbd5d471783f49dc10f7/store/tikv/2pc.go#L504-L516

Reply:
Or we should move the scatter operations into `WaitScatterRegionFinish`?
I'm not sure what will happen in `KV RPC Sender` if the split is succeeded but scatter failed

Reply:
Checked PDError in WaitScatterRegionFinish.
PTAL